devices.js

```diff
@@ -9073,6 +9073,22 @@ const devices = [
         description: 'LED A60 Zigbee RGB lamp',
         extend: preset.light_onoff_brightness_colortemp_colorxy,
     },
+    {
+        zigbeeModel: ['Smart Wall Switch '], // Yes, it has a space at the end :(
+        model: '421782',
+        vendor: 'Calex',
+        description: 'Smart Wall Switch, wall mounted RGB controller',
+        toZigbee: [],
+        fromZigbee: [fz.command_off, fz.command_on, fz.command_step, fz.command_move_to_color_temp,
+            fz.command_move, fz.command_stop, fz.command_ehanced_move_to_hue_and_saturation,
+        ],
+        exposes: [e.action([
+            'on', 'off', 'color_temperature_move', 'brightness_step_up', 'brightness_step_down',
+            'brightness_move_up', 'brightness_move_down', 'brightness_stop',
+            'enhanced_move_to_hue_and_saturation',
+        ])],
+        meta: {configureKey: 1, disableActionGroup: true},
```

Comment:
Battery shouldn't be difficult to support, add:

```js
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint = device.getEndpoint(1);
            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
            await reporting.batteryPercentageRemaining(endpoint);
        },
```

and add `e.battery()` to `exposes`.

Next start z2m, use the frontend and click the reconfigure button (or via MQTT: https://www.zigbee2mqtt.io/information/mqtt_topics_and_message_structure.html#zigbee2mqttbridgerequestdeviceconfigure). Check logs if OK, if you get timeout errors wakeup the device right before doing the configure. Next wait max 24 hours till battery reports.

Reply:
I don't think it has that endpoint. If I'm correct the `genPowerCfg` should also be in the endpoints right?

![image](https://user-images.githubusercontent.com/1292510/103483686-45e0a100-4de9-11eb-9e12-9be2bab371a1.png)

It also failed to bind that endpoint:
```plain
zigbee_1   | Zigbee2MQTT:error 2021-01-03 17:27:35: Failed to configure 'boven/werkkamer/kleur-schakelaar', attempt 3 (Error: Bind 0x00158d00024f3b6a/1 genPowerCfg from '0x00124b001caaac3d/1' failed (AREQ - ZDO - bindRsp after 10000ms)
zigbee_1   |     at Timeout._onTimeout (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:46:35)
zigbee_1   |     at listOnTimeout (internal/timers.js:554:17)
zigbee_1   |     at processTimers (internal/timers.js:497:7))
```

Reply:
Looks that it doesn't support it indeed.